src/middlewared/middlewared/plugins/network.py

```diff
@@ -1148,19 +1176,33 @@ async def _common_validation(self, verrors, schema_name, data, itype, update=Non
                                 f'{str(validation_attrs[i][0]) + str(validation_attrs[i][2])}',
                             )
 
-                # can't remove VHID and not GROUP
-                if not data['failover_vhid'] and data['failover_group']:
-                    verrors.add(
-                        f'{schema_name}.failover_vhid',
-                        'Removing only the VHID is not allowed.'
-                    )
+                if osc.IS_FREEBSD:
+                    # can't remove VHID and not GROUP
+                    if not data['failover_vhid'] and data['failover_group']:
+                        verrors.add(
+                            f'{schema_name}.failover_vhid',
+                            'Removing only the VHID is not allowed.'
+                        )
 
-                # can't remove GROUP and not VHID
-                if not data['failover_group'] and data['failover_vhid']:
-                    verrors.add(
-                        f'{schema_name}.failover_group',
-                        'Removing only the failover group is not allowed.'
-                    )
+                    # can't remove GROUP and not VHID
+                    if not data['failover_group'] and data['failover_vhid']:
+                        verrors.add(
+                            f'{schema_name}.failover_group',
+                            'Removing only the failover group is not allowed.'
+                        )
+
+                    if update.get('failover_vhid') != data['failover_vhid']:
+                        used_vhids = set()
+                        for v in (await self.middleware.call(
+                            'interface.scan_vrrp', data['name'], None, 5,
+                        )).values():
+                            used_vhids.update(set(v))
+                        if data['failover_vhid'] in used_vhids:
+                            used_vhids = ', '.join([str(i) for i in used_vhids])
+                            verrors.add(
+                                f'{schema_name}.failover_vhid',
+                                f'The following VHIDs are already in use: {used_vhids}.'
+                            )
```

Comment:
Doesn't Linux keepalived also uses VRRP packets or does it have its own conflict mechanisms?

Reply:
Great question, I've made it so that VRRP doesn't use multicast address space. It uses unicast, so the VHID is irrelevant 😄